ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGen/policyBuilder.go

```diff
@@ -42,17 +44,20 @@ func (pbuilder *PolicyBuilder) Build(policyGenTemp utils.PolicyGenTemplate) (map
 			if sFile.PolicyName == "" {
 				// Generate CR without append to policy
 				if len(resources) > 1 {
-					return policies, errors.New("Update spec/data of multiple yamls structure in same file " + sFile.FileName +
-						" not allowed. Instead separate them in multiple files")
-				}
-				resource := resources[0]
-				name := resource["kind"].(string)
-				name = name + "-" + resource["metadata"].(map[string]interface{})["name"].(string)
-
-				if resource["metadata"].(map[string]interface{})["namespace"] != nil {
-					name = name + "-" + resource["metadata"].(map[string]interface{})["namespace"].(string)
+					// Multi-document yaml- use the source filename
+					name = "Multiple-" + strings.TrimSuffix(sFile.FileName, filepath.Ext(sFile.FileName))
+					policies[utils.CustomResource+"/"+policyGenTemp.Metadata.Name+"/"+name] = resources
+				} else {
+					// Single-resource yaml - bake a filename based in the contents
+					resource := resources[0]
+					name := resource["kind"].(string)
+					name = name + "-" + resource["metadata"].(map[string]interface{})["name"].(string)
+
+					if resource["metadata"].(map[string]interface{})["namespace"] != nil {
+						name = name + "-" + resource["metadata"].(map[string]interface{})["namespace"].(string)
+					}
+					policies[utils.CustomResource+"/"+policyGenTemp.Metadata.Name+"/"+name] = resource
```

Comment:
same

Reply:
I can't do that; It's `go format` standard.  I can change it to a `fmt.Sprintf`, or maybe a `path.Join(...)` would be even better than string concatenation.